### PR TITLE
feat: Backport ros1 and ros2 targets from Avular

### PR DESCRIPTION
**Pull request by ar13pit**
Merge before https://github.com/tue-robotics/tue-env/pull/620.

CI pipeline of 620 is dependent on this

### CODE REVIEW

**Comment by ar13pit** [comment]
This PR implements half of https://github.com/tue-robotics/tue-env/issues/574

File: colcon/install.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- type: system
```
Comment (MatthijsBurgh) [inline colcon/install.yaml:1]: I think this requires either ros-setup or ros2-setup

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: ros2-setup/install.bash [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        tue-install-debug "Updating expired GPG key of ROS2 repository"
+        ADD_ROS_GPG_KEY=true
+    else
+        tue-install-debug "Not updating the existing GPG of the ROS2 repository"
+    fi
+fi
+
+if [[ "${ADD_ROS_GPG_KEY}" == "true" ]]
+then
+    sudo curl -sSL "${keyurl}" -o "${keyfile}"
```
Comment (MatthijsBurgh) [inline ros2-setup/install.bash:46]:
```suggestion
    tue-install-pipe sudo curl -sSL "${keyurl}" -o "${keyfile}"
```

File: ros2-setup/install.bash [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+then
+    echo "${sourceurl}" | sudo tee "${sourcefile}" > /dev/null
+    tue-install-debug "Successfully added/updated ROS2 source file in apt"
+
+    FORCE_UPDATE=true
+fi
+
+if [[ "${FORCE_UPDATE}" == "true" ]]
+then
+    sudo apt-get update -qq
```
Comment (MatthijsBurgh) [inline ros2-setup/install.bash:62]:
```suggestion
    tue-install-pipe sudo apt-get update -qq
```

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: colcon/install.yaml (hunk 1 above)
Comment (ar13pit) [reply colcon/install.yaml:4]: Done

File: ros2-setup/install.bash [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    tue-install-debug "ROS2 sources already added to apt-get"
+fi
+
+if [[ "${ADD_ROS_GPG_KEY}" == "false" ]]
+then
+    if [[ ! -f "${keyfile}" ]]
+    then
+        tue-install-debug "No existing GPG key of ROS2 repository found, adding a new one"
+        ADD_ROS_GPG_KEY=true
+    elif gpg --import --import-options show-only "${keyfile}" 2> /dev/null | grep -q expired
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply ros2-setup/install.bash:35]: This is also not in the `ros-setup` target

[the remark this answers is not in the document]
Comment (ar13pit) [reply ros2-setup/install.bash:35]: Does this correctly return the return flag of the executed command?

**Review by MatthijsBurgh — DISMISSED** [review]
Please check my changes to `ros-setup`

**Review by MatthijsBurgh — APPROVED** [review]

**Pull request merged**